viewing_party/party.py

```diff
@@ -0,0 +1,150 @@
+
+def create_movie(movie_title, genre, rating):
+    movie = {}
+    if movie_title is None or genre is None or rating is None:
+        return None
+    else:
+        movie["title"] = movie_title
+        movie["genre"] = genre
+        movie["rating"] = rating    
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(janes_data, watched_movie):
+    for movie in janes_data["watchlist"]:
+        if movie == watched_movie or watched_movie == movie["title"]:
+            janes_data["watched"].append(movie)
+            janes_data["watchlist"].remove(movie)
```

Comment:
Be careful about the side effects of modifying a collection as you're iterating through it (watchlist). For this example, we're only removing a single instance of a movie, however, if there are multiple instances of that movie that could cause the rest of the items in the list to shift position and potentially skip over other items as the loop continues.

This case is fine, but let's proceed with caution when modifying a collection while iterating through it.

https://thispointer.com/python-remove-elements-from-a-list-while-iterating/